Automattic/wp-calypso
# Coming soon: remove flag and permanently enable v2 for site creation

### Changes proposed in this Pull Request

Coming Soon (public not-indexable by default) has been in production and running smoothly. By "smoothly" we mean no serious bug reports, user difficulties or planetary-level implosions. 

This PR will be one in a series that removes the `coming-soon-v2` flag from Calypso and, eventually, removes the flag from the config files altogether. The consequence is that we are permanently excising v1 (private by default).

**This PR removes the coming soon v2 flag for site creation in both start and new onboarding flows. All new sites will therefore be created in Coming Soon v2™️ mode!**

### Testing instructions

1. Head over to `/start/domains` and create a site
2. In the POST request to `sites/new` check that the payload contains `public: 0` and `wpcom_public_coming_soon: 1`
3. Check that the site is in coming soon mode by visiting it in incognito mode.
4. Repeat for a site create via `/new`
5. Try making your own ketchup using tomato paste, salt, sugar and vinegar! It works!

<img width="1072" alt="coming-soon-v2-is-here" src="https://user-images.githubusercontent.com/6458278/100557604-5f7c5d80-32fe-11eb-952c-8a9d7ed4c7c0.png">
